### PR TITLE
Update desc for --no-progress

### PR DESCRIPTION
**Pull request by sanmai**

### CODE REVIEW

File: src/guide/command-line-options.md [hunk 1]
```diff
@@ -146,24 +146,27 @@ infection --mutators=PublicVisibility,Plus,Decrement,@number
...
 
 ```bash
-Creating mutated files and processes: 0/45678
-Creating mutated files and processes: 1/45678
-Creating mutated files and processes: 2/45678
-Creating mutated files and processes: 3/45678
+Processing source code files: 0/5678
+Processing source code files: 1/5678
+Processing source code files: 2/5678
+Processing source code files: 3/5678
```
Comment (sanmai) [inline src/guide/command-line-options.md]: `Creating mutated files and processes` is gone

File: src/guide/command-line-options.md [hunk 2]
```diff
@@ -146,24 +146,27 @@ infection --mutators=PublicVisibility,Plus,Decrement,@number
...
 ...
-Creating mutated files and processes: 45678/45678
+Processing source code files: 5678/5678
 
 ```
 
-Infection automatically enable this mode when either `CI` or `CONTINUOUS_INTEGRATION` environment variable are set to `"true"`.
-
-For example, there is no need to enable it manually on Travis CI.
+Progress bad display will be automatically disabled with or without this option when either `CI` or `CONTINUOUS_INTEGRATION` environment variables are set to `"true"`.
```
Comment (sanmai) [inline src/guide/command-line-options.md]: This is closer to the truth. I think it is better to not force this option on users, for time being.

File: src/guide/command-line-options.md [hunk 3]
```diff
@@ -146,24 +146,27 @@ infection --mutators=PublicVisibility,Plus,Decrement,@number
 
 ### `--no-progress`
 
-This option disables progress bars output. The main purpose is to use it on Continuous Integration servers to reduce the amount of generated text output, e.g.:
+This option has two effects:
+- It disables intermediate buffering of mutations used to count them. This causes progress bars to not have a total number of mutations displayed, while also reducing memory usage requirements and speeding up the entire process. It is benificial during CI, and for larger projects.
+- It disabled dynamic progress bars output to reduce the amount of generated text output.
```
Comment (sanmai) [inline src/guide/command-line-options.md]:
```suggestion
- It disables dynamic progress bars output to reduce the amount of generated text output.
```

File: src/guide/command-line-options.md [hunk 4]
```diff
@@ -146,24 +146,27 @@ infection --mutators=PublicVisibility,Plus,Decrement,@number
 
 ### `--no-progress`
 
-This option disables progress bars output. The main purpose is to use it on Continuous Integration servers to reduce the amount of generated text output, e.g.:
+This option has two effects:
+- It disables intermediate buffering of mutations used to count them. This causes progress bars to not have a total number of mutations displayed, while also reducing memory usage requirements and speeding up the entire process. It is benificial during CI, and for larger projects.
```
Comment (sanmai) [inline src/guide/command-line-options.md]:
```suggestion
- It disables intermediate buffering of mutations used to count them. This causes progress bars to not have a total number of mutations displayed, while also reducing memory usage and speeding up the entire process. It is beneficial during CI, and for larger projects.
```

File: src/guide/command-line-options.md [hunk 5]
```diff
@@ -146,24 +146,27 @@ infection --mutators=PublicVisibility,Plus,Decrement,@number
...
 ...
-Creating mutated files and processes: 45678/45678
+Processing source code files: 5678/5678
 
 ```
 
-Infection automatically enable this mode when either `CI` or `CONTINUOUS_INTEGRATION` environment variable are set to `"true"`.
-
-For example, there is no need to enable it manually on Travis CI.
+Progress bad display will be automatically disabled with or without this option when either `CI` or `CONTINUOUS_INTEGRATION` environment variables are set to `"true"`.
```
Comment (maks-rafalko) [inline src/guide/command-line-options.md]: `bad` -> `bar`

Comment (maks-rafalko) [inline src/guide/command-line-options.md]:
```suggestion
Progress bar display will be automatically disabled with or without this option when either `CI` or `CONTINUOUS_INTEGRATION` environment variables are set to `"true"`.
```

**Comment by maks-rafalko** [comment]
Thank you @sanmai

**Pull request merged**